internal/testutils/ethereum.go

```diff
@@ -0,0 +1,14 @@
+package testutils
+
+import (
+	"crypto/ecdsa"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func PrivateKeyFixture(t *testing.T) *ecdsa.PrivateKey {
```

Comment:
[Doc] Would be nice to add a one liner description for `PrivateKeyFixture'.

```suggestion
// PrivateKeyFixture generates a new random private key for use in tests.
// It fails the test immediately if key generation does not succeed.
func PrivateKeyFixture(t *testing.T) *ecdsa.PrivateKey {
```